warehouse/scripts/publish.py

```diff
@@ -477,14 +477,19 @@ def _publish_exposure(
 
                     if publish:
                         typer.secho(publish_msg, fg=typer.colors.GREEN)
-                        with open(fpath, "rb") as fp:
-                            upload_to_ckan(
-                                url=destination.url,
-                                fname=fname,
-                                fsize=fsize,
-                                file=fp,
-                                resource_id=resource.id,
-                            )
+                        try:
+                            with open(fpath, "rb") as fp:
+                                upload_to_ckan(
+                                    url=destination.url,
+                                    fname=fname,
+                                    fsize=fsize,
+                                    file=fp,
+                                    resource_id=resource.id,
+                                )
+                        except requests.exceptions.RequestException as e:
+                            print(e)
```

Comment:
We're already using `typer.echo` statements instead of the built-in `print` tool elsewhere in this script, so I'd make this into something like:

`typer.echo(f"Failed to upload to {fpath} due to error: {e}")`

We use styled logs in this script (which don't actually display with styling in Airflow, so jury's out on whether it's worth continuing to do so). But if you want to align this with most of the other logging in the same script, you'd use the slightly more complex `typer.secho` command like this one:

`typer.echo(f"Failed to upload to {fpath} due to error: {e}", fg=typer.colors.RED)`

Reply:
ah yes good point!